### DISCUSSION TITLE
**Please** help fix Xcode "No such module 'Capacitor'" - Detailed

### DISCUSSION THREAD

**Discussion by Sloris** [ionic-team/capacitor#4229] [General]
Hello,

I am not a programmer, to begin with, so I've spent the past 6 hours learning the fundamentals of Xcode and whatnot, but I am a tech-savvy autodidact. I am currently using the latest Xcode. I had a developer make me an app, and I'm trying to get it on the App Store now, but when I run my build I get the error in AppDelegate.swift:

`import Capacitor `["Error: No such module 'Capacitor'"]

I've tried various CL solutions and **nothing is working**. I am trying both the .xcodeproj and .xcworkspace files. Each has different errors. Errors for the [.xcodeproj](https://imgur.com/a/VjXrxBQ) and errors for the [.xcworkspaces](https://imgur.com/a/P3rbYHt). 

As you can see, on .xcodeproj the error is only 1 error in which there is No Capacitor module. If you look at .xcworkspace's errors, it's that **it cant open or locate the 3 Capacitor folders, even though they're there just like all the other pods are**. [Here is just a few of the cli things I've tried](https://imgur.com/a/Z6CRwf8). Also tried:

>[npm i (@)capacitor/cli](https://imgur.com/a/fixMqUl)

>[npm i (@)capacitor/ios](https://imgur.com/a/fixMqUl)

>pod install (i *think* this one worked?)

Please help me out. Thank you so much.

**Comment by mlynch** [comment]
Two things to check:

Make sure you ran `npx cap sync` which will have the iOS project install all of it's native dependencies

Then, make sure you run `npx cap open ios` or open the `.xcworkspace` file, *not* the `.xcodeproj` file. This one is important, only the workspace will have the pods resolved.

Try that and then let me know if that worked.

**Reply by Sloris** [reply]
Thanks so much for the response, I tried to run `npx cap sync` but recieved "command not found: cap"

In addition, my developer told me that inside my source folder was another folder labeled "ios", and to replace the ios folder ive been working with all this time on Xcode with this new similar one. I did this, ran it on Xcode, and the Capacitor doesn't seem to be an issue, but [now it's telling me this??](https://imgur.com/a/jyKiOVk) - Thanks again.

**Reply by mlynch** [reply]
If you get `command not found: cap`, then you need to run `npm install` in the root of your app project (the one where `package.json` lives). Then try the commands again.

Not following the second part or what that error is, sorry. If you haven't made any native code modifications (custom native code/plugins/etc.) you could remove the ios folder and re-create it using `npx cap add ios` and try the `npx cap sync` again.

**Reply by Sloris** [reply]
For the second part, I essentially am now working with a new but almost identical Xcode project/source. It no longer gives me my initial Capacitor errors, [but now it shows this](https://imgur.com/a/rZWb9gH), which I've enlarged so you can see what the errors are saying. I don't know how to fix this.

**Reply by mlynch** [reply]
Did you run those commands? Can you share the output?

**Reply by Sloris** [reply]
Here, it sort of worked (?) but it opened up a completely fresh workspace, and I need it in my already built Xcode project.
````
____________@____________-Mac-Pro App % npx cap sync
[error] Capacitor could not find the web assets directory "/Users/____________/Desktop/Equestock/Source v0.3/ios (2)/App/www".
    Please create it and make sure it has an index.html file. You can change
    the path of this directory in capacitor.config.json (webDir option).
    You may need to compile the web assets for your app (typically 'npm run build').
    More info: https://capacitorjs.com/docs/basics/building-your-app
____________@____________-Mac-Pro App % npx cap sync
[error] The web directory (/Users/____________/Desktop/Equestock/Source v0.3/ios (2)/App/www) must contain a "index.html".
    It will be the entry point for the web portion of the Capacitor app.
____________@____________-Mac-Pro App % npx cap sync
✔ Copying capacitor.js to web dir in 8.20ms
✔ copy in 12.85ms
✔ update web in 54.25μp
Sync finished in 0.027s
____________@____________-Mac-Pro App % npm install
npm WARN equestock@1.1.0 No repository field.

audited 60 packages in 1.098s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

____________@____________-Mac-Pro App % npx cap open ios 
[error] ios" platform has not been created. Use "npx cap add ios" to add the platform project.
____________@____________-Mac-Pro App % npx cap add ios
ℹ Installing iOS dependencies – Skipping: already installed
✔ Adding native xcode project in: /Users/____________/Desktop/Equestock/Source v0.3/ios (2)/App/ios in 31.94ms
✔ add in 33.59ms
✔ Copying web assets from www to ios/App/public in 4.93ms
✔ Copying native bridge in 2.20ms
✔ Copying capacitor.config.json in 1.93ms
✔ copy in 15.64ms
✔ Updating iOS plugins in 1.06ms
  Found 0 Capacitor plugins for ios:
✔ Updating iOS native dependencies with "pod install" (may take several minutes) in 6.58s
✔ update ios in 6.59s

Now you can run npx cap open ios to launch Xcode
____________@____________-Mac-Pro App % npx cap open ios
✔ Opening the Xcode workspace... in 3.01s
____________@____________-Mac-Pro App % cd
____________@____________-Mac-Pro ~ % cd /Users/____________/Desktop/Equestock/Source\ v0.3/ios\ \(2\) ____________@____________-Mac-Pro ios (2) % npx cap open equestock.xcodeproj
[error] platform equestock.xcodeproj not found
````

edit: opened up the .xcworkspace, [full of errors](https://imgur.com/a/NPgZOy0) that dont make sense. the files that are claimed to not be there are indeed there.

**Reply by mlynch** [reply]
I haven't seen that error before but I searched and this looks relevant: https://forum.ionicframework.com/t/app-pods-target-support-files-pods-app-pods-app-debug-xcconfig-unable-to-open-file-in-target-app-in-project-app/173459

Looks like it could be a permission issue. Perhaps try that suggestion in there and see if it resolves it

**Reply by Sloris** [reply]
It fixed it! **Almost!** I now have only two errors left, which I am still unsure of how to solve since [the errors saying the files aren't there](https://imgur.com/a/NBJXcJ6) definitely are. Need to fix this :(

**Edit:** Spoke too soon. I hit Run and got 8 [new/same errors](https://imgur.com/a/hXvoH4u). I don't know what's going on, it's so frustrating, but I'll do anything to fix it.

**Reply by mlynch** [reply]
Those just look like warnings, but something you could try at this point is doing a project clean in the build drop down, and then also clearing intermediate files: https://stackoverflow.com/a/39495772
